src/metabase/models/card.clj

```diff
@@ -594,29 +674,34 @@ saved later when it is ready."
   the transaction yet. If you pass true here it is important to call the event after the cards are successfully
   created."
   ([card creator] (create-card! card creator false))
-  ([{:keys [dataset_query result_metadata dataset parameters parameter_mappings], :as card-data} creator delay-event?]
-   ;; `zipmap` instead of `select-keys` because we want to get `nil` values for keys that aren't present. Required by
-   ;; `api/maybe-reconcile-collection-position!`
+  ([{:keys [dataset_query result_metadata dataset parameters parameter_mappings type] :as card-data} creator delay-event?]
+   (assert-card-type-and-dataset card-data)
    (let [data-keys            [:dataset_query :description :display :name :visualization_settings
-                               :parameters :parameter_mappings :collection_id :collection_position :cache_ttl]
-         card-data            (assoc (zipmap data-keys (map card-data data-keys))
-                                     :creator_id (:id creator)
-                                     :dataset (boolean (:dataset card-data))
-                                     :parameters (or parameters [])
-                                     :parameter_mappings (or parameter_mappings []))
+                               :parameters :parameter_mappings :collection_id :collection_position :cache_ttl :type]
+         ;; `zipmap` instead of `select-keys` because we want to get `nil` values for keys that aren't present. Required by
+         ;; `api/maybe-reconcile-collection-position!`
+         card-data            (-> (zipmap data-keys (map card-data data-keys))
+                                  (assoc
+                                   :dataset    (or (when (= "model" type)
```

Comment:
this `(or ...` logic is needed because we want to be able to create a model by using `type=model` or `dataset=true`

Reply:
This code is redundant and duplicates logic with `ensure-type-and-dataset-are-consistent`, similar to [this](https://github.com/metabase/metabase/pull/38074#discussion_r1475073288) comment.

Since you're threading this map into `ensure-type-and-dataset-are-consistent` you can just remove it with no change in behaviour.

Reply:
without the default value, i'll need to add `:dataset` to the list of `data-keys`, but in that case the `card-data` won't have both `dataset` and `type` key, which will eventually throw an error because there is no default value for those keys.

Reply:
Ah, you want to preserve the default if `type` or `dataset` aren't provided in the args? In that case you can explicitly handle that with a conditional that doesn't repeat the logic inside `ensure-type-and-dataset-are-consistent`. 

e.g.
```clojure
(-> (zipmap data-keys (map card-data data-keys))
    (cond-> (and (nil? type) (nil? dataset))
      (assoc :type "question"))
    (assoc ....)
    card/ensure-type-and-dataset-are-consistent)
```

Reply:
okay I like your version better, updated in 598d469ba2